pytket/phir/sharding/shards2ops.py

```diff
@@ -39,17 +43,36 @@ def parse_shards_naive(
             # if there are more than 2 qubits used, treat them all as parallel sq ops
             # one qubit will just be a single sq op
             # 3 or more will be 3 or more parallel sq ops
+            # when iterating through qubits,
+            # map all the qubits to a unique id to prevent duplicates in placement
             if len(shard.qubits_used) != 2:  # noqa: PLR2004
                 for qubit in shard.qubits_used:
-                    op = qubit.index
+                    qid, qubits2ids, qubit_id = qubit_id_helper(
+                        qubit, qubits2ids, qubit_id
+                    )
+                    op = [qid]
                     layer.append(op)
             else:
                 for qubit in shard.qubits_used:
-                    op.append(qubit.index[0])
+                    qid, qubits2ids, qubit_id = qubit_id_helper(
+                        qubit, qubits2ids, qubit_id
+                    )
+                    op.append(qid)
                 layer.append(op)
 
             scheduled.add(shard.ID)
 
         layers.append(layer)
 
     return layers, shards_in_layer
+
+
+def qubit_id_helper(
```

Comment:
So its good to have factored this function out to avoid repeating it, but a "helper function" is a _type_ of function, not a good name. This function should be named for what it actually _accomplishes_ and will mostly likely start with a verb of some sort since functions perform some kind of action.

Reply:
i renamed it to `assign_qubit_id`